sdk_commons/sdk_commons/config.py

```diff
@@ -0,0 +1,103 @@
+import abc
+from enum import Enum
+
+
+class SDKConfig(abc.ABC):
+    """
+    This is the abstract common interface that every venues
+    are expended to implement. Each method will be invoked by
+    API providing all listed parameters but each venue is free
+    to only include needed parameters in their own concrete
+    implementation. All other parameters will be consumed by
+    *args and **kwargs that are always expected to be included
+    A template venue is also provided as example for a concrete
+    implementation.
+    """
+
+    @property
+    @abc.abstractmethod
+    def authorization_pages(self):
+        """
+        Set this property with a class containing
+        mainnet and testnet properties
+        """
+        pass
+
+    @abc.abstractmethod
+    def venue_chains(self) -> Enum:
+        """
+        Return an Enum listing all supported chains
+        """
+        pass
+
+    @abc.abstractmethod
+    def create_offer(
+        self,
+        address: str,
+        chain_id: int,
+        rpc_uri: str,
```

Comment:
I think we could receive the parameters needed to instantiate the chain client (rpc_uri, chain_id) in the constructor, so we don't need to pass it to all methods

Reply:
Added a constructor on the base class to receive these parameters, the modified all the methods by removing them.
To contract config is now created as:
```
config = ContractConfig(address=self.address, chain_id=self.chain_id, rpc_uri=self.rpc_uri)
```

Probably each SDK Config could autonomously override the constructor to create che config once and then directly use a self.config. But for the moment I would like to focus on the interface and not on the specific SDK implementation